docs/sphinx/user_guide/feature/plugins.rst

```diff
@@ -0,0 +1,97 @@
+.. ##
+.. ## Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+.. ## and other RAJA project contributors. See the RAJA/COPYRIGHT file
+.. ## for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+.. ##
+
+.. _plugins-label:
+
+========
+Plugins
+========
+
+RAJA supports user-made plugins that may be loaded either at the time of compilation or during runtime. These two methods are not mutually exclusive, as plugins loaded dynamically can be run alongside plugins that are loaded statically.
+
+------------
+Plugin API
+------------
+
+
+^^^^^^^^^^^
+Required Functions
+^^^^^^^^^^^
+The preLaunch and postLaunch functions are automatically called by RAJA before and after loop execution. This applies to RAJA's kernel and forall implementations.
+
+* ``void preLaunch(PluginContext& p) {}`` - Will occur before kernel/forall execution.
+
+* ``void postLaunch(PluginContext& p) {}`` - Will occur after kernel/forall execution.
+
+^^^^^^^^^^^
+Unrequired Functions
+^^^^^^^^^^^
+The init and finalize functions have standard implementations and thus are not required to be included in a user-made plugin. Init and finalize are never run by RAJA by default and are only run when the user makes a call to RAJA::util::init_plugin() or RAJA::util::finalize_plugin() respectively.
+
+* ``void init(PluginOptions p) {}`` - Called by the user
+
+* ``void finalize() {}``
+
+
+-----------------
```

Comment:
Please add an explanation about how static and dynamic loading work differently and when to prefer one over the other.